### DISCUSSION TITLE
Extracting Query Parameters in Next.js App Router within a Layout.tsx Server Component

### DISCUSSION THREAD

**Discussion by ammar-bay** [vercel/next.js#62978] [App Router]
Why is there no way to get query params in layout component if the layout component is a server component.

I have this layout component and I want to get query param id from url such as localhost:3000/?id=123. But I could not find anyway to this. I want to make a decision based on the id and want to query db and it would be much easier if I could get the query param. I have gone through all the possible solution but could find on for my usecase.

`
import { redirect } from "next/navigation";
import { auth } from "@clerk/nextjs";

import prismadb from "@/lib/prismadb";

export default async function SetupLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  const { userId } = auth();

  if (!userId) {
    redirect("/sign-in");
  }

  const store = await prismadb.store.findFirst({
    where: {
      userId,
    },
  });

  if (store) {
    redirect(`/${store.id}`);
  }

  return <>{children}</>;
}
`

**Comment by moussaab-moulim** [comment]
it is not possible because Layout is not rerendered when page change
check documentation
you do that in page
https://nextjs.org/docs/app/api-reference/functions/use-search-params#server-components
